modules/core/src/main/java/org/apache/ignite/internal/processors/localtask/DurableBackgroundTasksProcessor.java

```diff
@@ -99,10 +123,12 @@ private void asyncDurableBackgroundTaskExecute(DurableBackgroundTask task, boole
                 catch (Throwable e) {
                     log.error("Could not execute durable background task: " + task.shortName(), e);
 
-                    if (dropTaskIfFailed)
-                        removeDurableBackgroundTask(task);
+                    if (e instanceof AssertionError)
```

Comment:
I believe any java.lang.Error should be considered as critical because it leads to Worker crash with unpredictable behavior of the whole system.

Reply:
I agree with you, I fixed it.